src/main/java/web/service/ChessService.java

```diff
@@ -0,0 +1,111 @@
+package web.service;
+
+import chess.domain.board.Board;
+import chess.domain.board.InitialBoardGenerator;
+import chess.domain.board.LineNumber;
+import chess.domain.board.Point;
+import chess.domain.game.GameState;
+import chess.domain.game.Ready;
+import chess.domain.piece.*;
+import chess.request.BoardAndTurnInfo;
+import chess.request.ScoreResponse;
+import web.dao.BoardDao;
+import web.dao.GameDao;
+import web.dto.*;
+
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+public class ChessService {
+
+    private GameState gameState;
+    private final BoardDao boardDao;
+    private final GameDao gameDao;
+
+    public ChessService(BoardDao boardDao, GameDao gameDao) {
+        this.gameState = new Ready();
+        this.boardDao = boardDao;
+        this.gameDao = gameDao;
+    }
+
+    public WebBoardDto startNewGame(GameDto gameDto) {
+        gameDao.save(gameDto);
+        gameState = this.gameState.start(new Board(InitialBoardGenerator.generate()), Color.WHITE);
+
+        BoardAndTurnInfo response = (BoardAndTurnInfo) gameState.getResponse();
+        Map<Point, Piece> responseBoard = response.getBoard();
+        for (Point point : responseBoard.keySet()) {
+            savePiece(gameDto, responseBoard, point);
+        }
+        return new WebBoardDto(response, true);
+    }
+
+    private void savePiece(GameDto gameDto, Map<Point, Piece> responseBoard, Point point) {
+        if (!responseBoard.get(point).isSameType(PieceType.EMPTY)) {
+            boardDao.save(new PieceDto(gameDto.getRoomName(), point.convertPointToId(),
+                    responseBoard.get(point).getPieceType(), responseBoard.get(point).getPieceColor()));
+        }
+    }
+
+    public WebBoardDto resumeGame(GameDto gameDto) {
+        GameDto findGame = gameDao.findByRoomName(gameDto.getRoomName());
+        if (findGame == null) {
+            throw new IllegalArgumentException("[ERROR] 존재하는 게임이 없습니다.");
+        }
+        List<PieceDto> allPieces = boardDao.findAllPiecesByRoomName(gameDto.getRoomName());
+        Map<Point, Piece> pointPieces = new HashMap<>();
+        createBoard(pointPieces, allPieces);
```

Comment:
```suggestion
        Map<Point, Piece> pointPieces = createBoard(allPieces);
```

자료 구조를 넘겨서 채워주는 것보다는 불변 자료구조를 반환받아서 할당하는 편이 가독성이 좋습니다 :)

Reply:
범블비의 말대로 해당 내용을 수정해보니 가독성 측면에서 훨씬 좋아보이는 것 같네요!

앞으로도 자료구조를 넘겨서 채워주기보다 반환받아 할당하도록 하겠습니다😁